<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CalibrationModule</class>
 <widget class="QWidget" name="CalibrationModule">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>标定计算</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QGroupBox" name="calibrationSettingsGroup">
       <property name="title">
        <string>标定设置</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="1" column="0">
         <widget class="QLabel" name="labelBoardWidth">
          <property name="text">
           <string>Cols</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="boardWidthSpin">
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="value">
           <number>9</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelBoardHeight">
          <property name="text">
           <string>Rows</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="boardHeightSpin">
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="value">
           <number>6</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="labelSquareSize">
          <property name="text">
           <string>Size(mm):</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="squareSizeSpin">
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="value">
           <double>20.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="labelCalibrationType">
          <property name="text">
           <string>标定类型:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="calibrationTypeCombo">
          <item>
           <property name="text">
            <string>单目相机标定</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>双目相机标定</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>水下畸变校正</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="labelDistortionModel">
          <property name="text">
           <string>畸变模型:</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="distortionModelCombo">
          <item>
           <property name="text">
            <string>5参数模型</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>8参数模型</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>鱼眼相机模型</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <widget class="QCheckBox" name="undistortCheckBox">
          <property name="text">
           <string>计算去畸变映射</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="labelCalibrationType_2">
          <property name="text">
           <string>标定板类型:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="calibrationType">
          <item>
           <property name="text">
            <string>棋盘格</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>对称圆点</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>非对称圆点</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="progressGroup">
       <property name="title">
        <string>标定进度</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QProgressBar" name="calibrationProgressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="logTextEdit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="startCalibrationButton">
            <property name="text">
             <string>开始标定</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="cancelCalibrationButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>取消</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="saveParametersButton">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>保存参数</string>
            </property>
            <property name="checkable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="resultsGroup">
     <property name="title">
      <string>标定结果</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QTabWidget" name="resultsTabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="parametersTab">
         <attribute name="title">
          <string>内参矩阵</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QTableWidget" name="cameraMatrixTable">
            <property name="editTriggers">
             <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
            </property>
            <property name="rowCount">
             <number>3</number>
            </property>
            <property name="columnCount">
             <number>3</number>
            </property>
            <attribute name="horizontalHeaderMinimumSectionSize">
             <number>50</number>
            </attribute>
            <attribute name="horizontalHeaderDefaultSectionSize">
             <number>220</number>
            </attribute>
            <attribute name="verticalHeaderMinimumSectionSize">
             <number>50</number>
            </attribute>
            <attribute name="verticalHeaderDefaultSectionSize">
             <number>50</number>
            </attribute>
            <row/>
            <row/>
            <row/>
            <column/>
            <column/>
            <column/>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelReprojectionError">
            <property name="text">
             <string>平均重投影误差: --</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="distortionTab">
         <attribute name="title">
          <string>畸变系数</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QTableWidget" name="distortionTable">
            <property name="rowCount">
             <number>1</number>
            </property>
            <property name="columnCount">
             <number>5</number>
            </property>
            <attribute name="horizontalHeaderDefaultSectionSize">
             <number>100</number>
            </attribute>
            <attribute name="verticalHeaderMinimumSectionSize">
             <number>30</number>
            </attribute>
            <attribute name="verticalHeaderDefaultSectionSize">
             <number>30</number>
            </attribute>
            <row/>
            <column/>
            <column/>
            <column/>
            <column/>
            <column/>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="extrinsicsTab">
         <attribute name="title">
          <string>外参</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QListWidget" name="extrinsicsListWidget"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="previewLabel">
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>300</height>
         </size>
        </property>
        <property name="text">
         <string>标定结果预览</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
